Sources/core/objects/Human.cpp

```diff
@@ -82,6 +84,274 @@ bool Human::TryMove(Dir direct)
         {
             PlaySoundIfVisible("glass_step.ogg", GetOwner().Id());
         }
+        if (pulled_object_)
+        {
+            if (!CanTouch(pulled_object_))
```

Comment:
Maybe now it is implemented in this way, but previously it was in the way which I described. I do not see any pros of the new method, but only cons.
1. It decreases mobility of the pulled item (it will be much harder to pull items in thin tunnels)
2. You **must** have another action (e.g. double click pull) in order to compensate that. Sure, such feature actually is pretty neat, and it will be nice to have it someday - but you will **have** to use it. It is not conveniently.
3. It is much harder to implement it, because it requests diagonal movement in some way.

So I do not see reasons to make it in that way.